pymc_marketing/clv/models/basic.py

```diff
@@ -22,14 +21,27 @@ class CLVModel(ModelBuilder):
 
     def __init__(
         self,
+        data: Optional[pd.DataFrame] = None,
+        *,
         model_config: Optional[Dict] = None,
         sampler_config: Optional[Dict] = None,
     ):
         super().__init__(model_config, sampler_config)
+        self.data = data
 
     def __repr__(self):
         return f"{self._model_type}\n{self.model.str_repr()}"
 
+    def _add_fit_data_group(self, data: pd.DataFrame) -> None:
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                category=UserWarning,
+                message="The group fit_data is not defined in the InferenceData scheme",
+            )
+            assert self.idata is not None
```

Comment:
This `assert` statement will be skipped if python is ran in optimized mode. An IF statement raising a `RunTimeError` instead will prevent this.

Reply:
This is just for mypy

Reply:
Also did you ever see anyone running python in optimized mode :) ?

Reply:
`mypy` requires assert statements some of the typing (non)-issues it raises?

I could see a junior developer hearing that python has an optimized mode, getting excited & running a script without realizing literally all it does is skip `assert` lines, then getting stuck with an issue that could be quite hard to debug. Just a little extra guarantee for reliability.

Reply:
Yes, when mypy doesn't know about type information, you can convince him that all is right by adding an assert there. In here the issue in self.model is always None at first, and mypy doesn't know that we only call this method once it is no longer None. 

It's really just for mypy not for users sake that we have the assert